symengine/derivative.cpp

```diff
@@ -438,7 +438,25 @@ static RCP<const Basic> diff(const CLASS &self, \
             }
             return UnivariateIntPolynomial::from_dict(self.get_var(), std::move(d));
         } else {
-            return zero;
+            map_uint_mpz d;
+            d[0] = integer_class(0);
+            return UnivariateIntPolynomial::from_dict(self.get_var(), std::move(d));
+        }
+    }
+
+    static RCP<const Basic> diff(const UnivariatePolynomial &self,
+            const RCP<const Symbol> &x) {
+        if (self.get_var()->__eq__(*x)) {
+            map_int_Expr d;
+            for (const auto &p : self.get_dict()) {
+                d[p.first - 1] = p.second * p.first;
```

Comment:
@isuruf It seems to be failing when p.first, the degree, is 0. That's the case to check right?